include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -818,6 +839,8 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::FEPointEvaluation(
   , is_reinitialized(false)
 {
   setup(first_selected_component);
+  mapping_info.connect_is_reinitialized(
+    [this]() { this->is_reinitialized = false; });
```

Comment:
@bergbauer Have you addressed the issue that `FEPointEvaluation` might get out of scope and `MappingInfo` might want to update it? Or is this not an issue?

Reply:
To me this sounds like luck or is there any mechanism that deconnects?

Reply:
A small program that shows that shows that the current usage of the signal is not correct:
```
#include <boost/signals2.hpp>

#include <iostream>

struct A
{
  boost::signals2::signal<void()> signal;

  void
  reinit()
  {
    std::cout << "A::reinit()" << std::endl;
    signal();
  }
};

struct B
{
  B(A &a)
  {
    connection = a.signal.connect([]() { std::cout << "  B" << std::endl; });
  }

  ~B()
  {
    connection.disconnect(); // on master missing
  }

  boost::signals2::connection connection;
};

int
main()
{
  A a;
  a.reinit();

  {
    B b(a);
    a.reinit();
  }

  {
    B b(a);
    a.reinit();
  }

  a.reinit();
}
```
The solution is to store the connection and to disconnect at the end.